subsys/bluetooth/scan.c

```diff
@@ -791,6 +816,103 @@ static int scan_appearance_filter_add(u16_t appearance)
 	return 0;
 }
 
+static bool adv_manufacturer_data_cmp(const u8_t *data,
+									  u8_t data_len,
+									  const u8_t *target_data,
+									  u8_t target_data_len)
+{
+	if (target_data_len > data_len) {
+		return false;
+	}
+
+	for (size_t i = 0; i < target_data_len; i++) {
+		if (target_data[i] != data[i]) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static bool adv_manufacturer_data_compare(const struct bt_data *data)
+{
+	const struct bt_scan_manufacturer_data_filter *md_filter =
+		&bt_scan.scan_filters.manufacturer_data;
+	u8_t counter = bt_scan.scan_filters.manufacturer_data.cnt;
+
+	/* Compare the name found with the name filter. */
+	for (size_t i = 0; i < counter; i++) {
+		if (adv_manufacturer_data_cmp(data->data,
+				data->data_len,
+				md_filter->manufacturer_data[i].data,
+				md_filter->manufacturer_data[i].data_len)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static bool is_manufacturer_data_filter_enabled(void)
+{
+	return bt_scan.scan_filters.manufacturer_data.enabled;
+}
+
+static void manufacturer_data_check(struct bt_scan_control *control,
+									const struct bt_data *data)
```

Comment:
Remove this line-break.

Reply:
@joerchan: I see what the problem is. I used 4 space-tabs instead of 8, that's the reason why it looks so strange. But when I remove this line, the line will be > 80 chars. Should I instead remove the unnecessary tabs to align it with the struct?

Reply:
Ah right. That makes sense. So 9 tabs should become 4 tabs + 4 spaces then :)